### ISSUE TITLE
lib.discover_metadata_in_cyclus_path() segfaulting.

### ISSUE THREAD

**Issue by bam241** [cyclus/cyclus#1543]
```
>>> lib.discover_metadata_in_cyclus_path()                                                                            │Traceback (most recent call last): [1]    5152 segmentation fault (core dumped)  python
```

I seen that on an ubuntu 18.04 with python 3.6.8 cycles 1.5.5

**Comment by gonuke** [comment]
Is there any chance this is related to the problems you see on OSX?

**Comment by bam241** [comment]
Possible.
But I don't think so, it does not occur at the same time.

OS X issue occurs when importing Cyclus.

This one is on the `discover_metadata_in_cyclus_path()` call ...

**Comment by nuclearkatie** [comment]
I have this issue now too. Before messing with #1552, I probably hadn't re-installed cyclus since before this issue was opened. Now I get a segfault when using ubuntu 18.04 with python 3.6.9 cyclus 1.5.5 (almost identical to Baptiste)

Running `discover_metadata_in_cyclus_path()` is basically the very first thing that Trailmap does, so this effectively breaks Trailmap as well

**Comment by nuclearkatie** [comment]
I just confirmed that the issue does not occur in Cyclus 1.5.4

**Comment by gonuke** [comment]
I just pulled the cyclus docker image from last week and cycamore docker image from 1 month ago and it had no segfault.  We should figure out what combination is causing the problem.
The docker images are using python 3.7.3 on Debian 9.

**Comment by dermesser** [comment]
> I just pulled the cyclus docker image from last week and cycamore docker image from 1 month ago and it had no segfault. We should figure out what combination is causing the problem.
> The docker images are using python 3.7.3 on Debian 9.

I can reproduce this identically: cyclus works in the `cyclus/cyclus` Docker container, and it works when compiled with [Cython 0.28, but not with Cython 0.29](https://github.com/cyclus/cyclus/issues/1512#issuecomment-518908971). (this comment is mostly for others hitting the same bug)

Alas I'm not at all familiar with cython (yet)...

**Comment by gonuke** [comment]
I've asked @bennibbelink to confirm whether this is still an issue

**Comment by bennibbelink** [comment]
This is not an issue with the latest cyclus images.  It was likely fixed by #1666.  Was able to pull a `cyclus:latest` and run:
```python
import cyclus.lib
cyclus.lib.discover_metadata_in_cyclus_path()
```
and a JSON object of cyclus metadata was returned.  Closing this issue, if the segmentation faults return with a post-#1666 version of cyclus please open a new issue.
